dht/sharder.hh

```diff
@@ -88,12 +109,15 @@ struct ring_position_range_and_shard_and_element : ring_position_range_and_shard
 //    not, and which shards share it).
 //
 // During migration uses a view on shard routing for reads.
-class ring_position_range_vector_sharder {
-    using vec_type = dht::partition_range_vector;
+template <generalized_ring_position RingPositionLike>
+class generalized_ring_position_range_vector_sharder {
+public:
+    using vec_type = std::conditional_t<std::same_as<RingPositionLike, ring_position>, dht::partition_range_vector, dht::token_range_vector>;
+private:
```

Comment:
Looks like to unstall, we only need to parametrize on vec_type to use chunked_vector. 

The generalization to use tokens is not necessary.